File: app/src/main/java/com/br/example/fakebank/presentations/viewModels/viewModelFactories/StatementViewModelFactory.java
package com.br.example.fakebank.presentations.viewModels.viewModelFactories;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.br.example.fakebank.domains.services.StatementService;
import com.br.example.fakebank.presentations.handles.StatementHandle;
import com.br.example.fakebank.presentations.viewModels.StatementViewModel;

public class StatementViewModelFactory implements ViewModelProvider.Factory {

    private StatementService currencyService;
    private StatementHandle statementHandle;

    public StatementViewModelFactory(StatementService currencyService, StatementHandle statementHandle) {
        this.currencyService = currencyService;
        this.statementHandle = statementHandle;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(StatementViewModel.class)){
            return (T) new StatementViewModel(currencyService, statementHandle);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: app/src/main/java/com/gustavo/bankandroid/statements/viewmodel/StatementViewModelFactory.kt
package com.gustavo.bankandroid.statements.viewmodel

import android.content.Context
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.gustavo.bankandroid.statements.injection.StatementInjection

@Suppress("UNCHECKED_CAST")
class StatementViewModelFactory(private val context: Context) : ViewModelProvider.Factory {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return StatementViewModel(
            fetchStatementUseCase,
            getLoggedUserInfoUseCase
        ) as T
    }

    private val fetchStatementUseCase
        get() = (context as StatementInjection).fetchStatementUseCase
    private val getLoggedUserInfoUseCase
        get() = (context as StatementInjection).getLoggedUserInfoUseCase
}